oras/provider.py

```diff
@@ -840,6 +840,9 @@ def push(
         response = self.upload_manifest(
             manifest, container
         )  # make the returned response from this method, the one pertaining to the uploaded Manifest
+        if response.status_code == 500:
+            # retry once
```

Comment:
I wonder if instead of making an explicit retry here, we should check if this was meant to be captured by this decorator instead:

https://github.com/oras-project/oras-py/blob/ff9c2e28e364155de68bd653ddb4da9f6979d677/oras/decorator.py#L39